lib/messengers/http_messenger.ex

```diff
@@ -106,7 +107,8 @@ defmodule TxtLocalEx.HttpMessenger do
 
     case Request.request(:post, @message_status_path, sms_payload) do
       {:ok, response} -> response.body
-      {:error, error} -> raise TxtLocalEx.Errors.ApiError, error.reason
+      {:error, error} when is_binary(error) -> raise TxtLocalEx.Errors.ApiError, error
+      {:error, error} -> raise error
```

Comment:
Can you write a test to cover this?

Reply:
We don't need to handle this clause - 

```elixir
{:error, error} when is_binary(error) -> raise TxtLocalEx.Errors.ApiError, error
```

Removing it!